modules/gateway/gateway.go

```diff
@@ -146,10 +146,11 @@ type Gateway struct {
 	peerTG siasync.ThreadGroup
 
 	// Utilities.
-	log        *persist.Logger
-	mu         sync.RWMutex
-	persistDir string
-	threads    siasync.ThreadGroup
+	log         *persist.Logger
+	mu          sync.RWMutex
+	persistDir  string
+	threads     siasync.ThreadGroup
+	remotePeers int64
```

Comment:
why do we need `remotePeers`? Can't you calculate the value by looping through the peer list?

Reply:
I didn't want to loop through the whole peer map whenever we delete a key from the map

Reply:
you only loop when you call `Online` though, right? And that's not a call that we make frequently. Even if it were, locking a mutex and looping through <100 pointers isn't a slow operation by any means. Adding a new piece of mutable state introduces the possibility for `remotePeers` to desynchronize from the actual number of remote peers. It's an optimization that we would only make if calculating the value on-the-fly was very expensive.